ballerina/channel_csv_io.bal

```diff
@@ -40,30 +40,56 @@ Error {
     return (check getReadableCSVChannel(readableChannel, 0)).csvStream();
 }
 
-isolated function channelWriteCsv(WritableChannel writableChannel, string[][]|map<anydata>[] content) returns Error? {
-    WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
-    if content is string[][] {
-        foreach string[] r in content {
+isolated function channelWriteCsv(string path, FileWriteOption inputOption, string[][]|map<anydata>[] contentToWrite) returns Error? {
+    WritableCSVChannel csvChannel;
+    FileWriteOption option = OVERWRITE;
+    if contentToWrite is string[][] {
+        csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = inputOption));
+        foreach string[] r in contentToWrite {
             Error? writeResult = csvChannel.write(r);
             if writeResult is Error {
                 check csvChannel.close();
                 return writeResult;
             }
         }
-    } else if content is map<anydata>[] {
-        foreach map<anydata> row in content {
+        check csvChannel.close();
+    } else if contentToWrite is map<anydata>[] {
+        string[] headers = [];
+        if contentToWrite.length() == 0 {
+            return;
+        }
+        headers = contentToWrite[0].keys();
+        if inputOption == APPEND {
+            string[] headersFromCSV = check readHeadersFromCsvFile(path);
+            if headersFromCSV.length() > 0 {
+                headers = check validateCsvContent(headersFromCSV, headers);
```

Comment:
```suggestion
                headers = check validateCsvHeaders(headersFromCSV, headers);
```

Reply:
Updated in [65cb675](https://github.com/ballerina-platform/module-ballerina-io/pull/827/commits/65cb67563d075fbe62ca48feca1b11f8b97b4726)